Workday/canvas-kit
# feat: Add new `description` variant to Tooltip and deprecate `describe` variant for better accessibility.

## Summary

Fixes: #2797  




## Release Category
Components

### Release Note
- We've added a new `description` variant and **deprecated** the `describe` variant.

The `describe` variant of the tool tip relies on `aria-describedby` attribute to assign the tool tip text to the accessible description of the target element. This only works when the tool tip is visible and present in the DOM, which only occurs when the element has focus. Screen readers can access web page elements without necessarily using keyboard focus, therefore, this variant didn't appear to always work reliably for screen reader users. 

The preferred type to use is `description` to provide better accessible support.

## Checklist

- [x] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)

## For the Reviewer



For the `describe` variant of `Tooltip`, I replaced the `aria-describedby` attribute with `aria-description` and set the `title` prop value string. Also, remove the check for if visible, so we can always assign this string to the description. 


## Where Should the Reviewer Start?



- in the `useTooltip` custom hook
- check out the updates to the Describe examples in storybook
- check out the new "Tooltip" section under "Examples" in Storybook

## Areas for Feedback? (optional)





## Testing Manually



In the following scenarios, validate the issues cannot be reproduced and the tool tip text is always described without focusing the button elements. 

Steps to reproduce the behavior:
With VoiceOver / MacOS:
1. Launch VoiceOver, open Safari
2. Canvas Design Site: Tooltip popups > [Describing an element](https://canvas.workdaydesign.com/components/popups/tooltip#tab=web&heading=describing-an-element)
3. Reading commands: Ctrl + Opt + Right
4. Issue: observe tool tip text on Delete button is not described
5. Press Tab to focus “Show Code”
6. Read backward: Ctrl + Opt + Left Arrow
7. Issue: observe tool tip text on Delete button is not described
8. Set Rotor to “Form Fields”: Ctrl + Opt + Cmd + Right Arrow
9. Jump buttons on the page: Ctrl + Opt + Cmd + UP / DOWN arrow
10. Issue: observe tool tip text on Delete button is not described

With NVDA on Windows 11:
1. Launch NVDA, open Chrome
2. Canvas site => Tooltips => [Describing an element](https://canvas.workdaydesign.com/components/popups/tooltip#tab=web&heading=describing-an-element)
3. In virtual mode, use quick key ‘b’ to jump to the Delete Button
4. Issue: observe the tool tip text is not described

## Screenshots or GIFs (if applicable)



No visual changes. Just `aria` attribute change. 

## Thank You Gif (optional)
